[BUG] v0.4.13 forget settings changes in IDEA

Hi,

I'm changing some settings in the Plugin in IDEA 2024.3.x Ultimate Edition

![Image](https://github.com/user-attachments/assets/19816fc3-aaf8-4aed-8617-cf4cd34d9e19)

![Image](https://github.com/user-attachments/assets/67f89117-858f-435e-a06c-45cb0f01d845)

But after restart the IDEA default settings are restored and my changes are gone.

![Image](https://github.com/user-attachments/assets/ef3e2234-0b22-46be-89cc-6a9572a3d1f6)

I reinstalled the plugin and installed it again. I repeated thix 3-4 times.

It's an IDEA problem or plugin probleum. Cause in PyCharm Community Edition it's working.

I see this
```
com.intellij.diagnostic.PluginException: Cannot init component state (componentName=com.devoxx.genie.ui.SettingsState, componentClass=DevoxxGenieStateService) [Plugin: com.devoxx.genie]
	at com.intellij.configurationStore.ComponentStoreImpl.initComponent(ComponentStoreImpl.kt:170)
	at com.intellij.configurationStore.ComponentStoreWithExtraComponents.initComponent(ComponentStoreWithExtraComponents.kt:46)
	at com.intellij.serviceContainer.ComponentManagerImpl.initializeService$intellij_platform_serviceContainer(ComponentManagerImpl.kt:657)
	at com.intellij.serviceContainer.ServiceInstanceInitializer.createInstance$lambda$1$lambda$0(ServiceInstanceInitializer.kt:69)
	at com.intellij.platform.instanceContainer.instantiation.InstantiateKt.withStoredTemporaryContext(instantiate.kt:328)
	at com.intellij.serviceContainer.ServiceInstanceInitializer.createInstance$suspendImpl(ServiceInstanceInitializer.kt:68)
	at com.intellij.serviceContainer.ServiceInstanceInitializer.createInstance(ServiceInstanceInitializer.kt)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1$1.invokeSuspend(LazyInstanceHolder.kt:163)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1$1.invoke(LazyInstanceHolder.kt)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1$1.invoke(LazyInstanceHolder.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:62)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:163)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1.invokeSuspend(LazyInstanceHolder.kt:161)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1.invoke(LazyInstanceHolder.kt)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1.invoke(LazyInstanceHolder.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startCoroutineUndispatched(Undispatched.kt:27)
	at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:90)
	at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:123)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:52)
	at kotlinx.coroutines.BuildersKt.launch(Unknown Source)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.initialize(LazyInstanceHolder.kt:146)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.access$initialize(LazyInstanceHolder.kt:14)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.tryInitialize(LazyInstanceHolder.kt:136)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.getInstance(LazyInstanceHolder.kt:96)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.getInstanceInCallerContext$suspendImpl(LazyInstanceHolder.kt:88)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.getInstanceInCallerContext(LazyInstanceHolder.kt)
	at com.intellij.serviceContainer.ComponentManagerImplKt$doGetOrCreateInstanceBlocking$1.invokeSuspend(ComponentManagerImpl.kt:1566)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:277)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:111)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$BuildersKt__BuildersKt(Builders.kt:84)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:52)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at com.intellij.serviceContainer.ComponentManagerImplKt.runBlockingInitialization$lambda$10(ComponentManagerImpl.kt:1684)
	at com.intellij.openapi.progress.ContextKt.prepareThreadContext(context.kt:85)
	at com.intellij.serviceContainer.ComponentManagerImplKt.runBlockingInitialization(ComponentManagerImpl.kt:1675)
	at com.intellij.serviceContainer.ComponentManagerImplKt.doGetOrCreateInstanceBlocking(ComponentManagerImpl.kt:1565)
	at com.intellij.serviceContainer.ComponentManagerImplKt.getOrCreateInstanceBlocking(ComponentManagerImpl.kt:1560)
	at com.intellij.serviceContainer.ComponentManagerImpl.doGetService(ComponentManagerImpl.kt:751)
	at com.intellij.serviceContainer.ComponentManagerImpl.getService(ComponentManagerImpl.kt:695)
	at com.devoxx.genie.ui.settings.DevoxxGenieStateService.getInstance(DevoxxGenieStateService.java:33)
	at com.devoxx.genie.service.ChatMemoryService.init(ChatMemoryService.java:37)
	at com.devoxx.genie.service.PostStartupActivity.execute(PostStartupActivity.java:19)
	at com.intellij.ide.startup.impl.StartupManagerImplKt$launchActivity$1.invokeSuspend(StartupManagerImpl.kt:493)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:608)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:873)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:763)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:750)
Caused by: com.intellij.util.xmlb.XmlSerializationException: Cannot deserialize class com.devoxx.genie.ui.settings.DevoxxGenieStateService
	at com.intellij.configurationStore.StateStorageBaseKt.deserializeStateWithController(StateStorageBase.kt:375)
	at com.intellij.configurationStore.StateStorageBase.getState(StateStorageBase.kt:55)
	at com.intellij.configurationStore.ComponentStoreImpl$doCreateStateGetter$1.getState(ComponentStoreImpl.kt:560)
	at com.intellij.configurationStore.ComponentStoreImpl.doInitComponent(ComponentStoreImpl.kt:492)
	at com.intellij.configurationStore.ComponentStoreImpl.initComponent(ComponentStoreImpl.kt:430)
	at com.intellij.configurationStore.ComponentStoreImpl.initComponent(ComponentStoreImpl.kt:145)
	... 51 more
Caused by: java.lang.IllegalArgumentException: No argument provided for a required parameter: parameter #0 arg0 of fun `<init>`(com.devoxx.genie.model.enumarations.ModelProvider!, kotlin.String!, kotlin.String!, kotlin.Boolean, kotlin.Double, kotlin.Double, kotlin.Int, kotlin.Int): com.devoxx.genie.model.LanguageModel
	at kotlin.reflect.jvm.internal.KCallableImpl.callDefaultMethod$kotlin_reflection(KCallableImpl.kt:195)
	at kotlin.reflect.jvm.internal.KCallableImpl.callBy(KCallableImpl.kt:112)
	at com.intellij.serialization.xml.KotlinAwareBeanBindingKt.createUsingKotlin(KotlinAwareBeanBinding.kt:133)
	at com.intellij.serialization.xml.KotlinAwareBeanBindingKt.access$createUsingKotlin(KotlinAwareBeanBinding.kt:1)
	at com.intellij.serialization.xml.KotlinAwareBeanBinding.resolveInstantiator$lambda$0(KotlinAwareBeanBinding.kt:34)
	at com.intellij.serialization.xml.KotlinAwareBeanBinding.newInstance(KotlinAwareBeanBinding.kt:119)
	at com.intellij.util.xmlb.BeanBinding.deserialize(BeanBinding.kt:196)
	at com.intellij.util.xmlb.CollectionBinding.deserializeItem$intellij_platform_util(CollectionBinding.kt:306)
	at com.intellij.util.xmlb.CollectionStrategyImpl.deserializeList(CollectionBinding.kt:343)
	at com.intellij.util.xmlb.CollectionBinding.deserializeList(CollectionBinding.kt:259)
	at com.intellij.util.xmlb.TagBinding.deserialize(TagBinding.kt:214)
	at com.intellij.util.xmlb.BeanBindingKt.deserializeNestedBindingInto(BeanBinding.kt:484)
	at com.intellij.configurationStore.StateStorageBaseKt.getXmlSerializedState(StateStorageBase.kt:452)
	at com.intellij.configurationStore.StateStorageBaseKt.deserializeStateWithController(StateStorageBase.kt:327)
	... 56 more
```
exception on IDEA start.

--- a/core/src/main/java/com/devoxx/genie/model/LanguageModel.java
+++ b/core/src/main/java/com/devoxx/genie/model/LanguageModel.java
@@ -1,14 +1,20 @@
 package com.devoxx.genie.model;
 
 import com.devoxx.genie.model.enumarations.ModelProvider;
+
+import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
+import lombok.NoArgsConstructor;
+
 import org.jetbrains.annotations.NotNull;
 
 import java.util.Comparator;
 
 @Data
 @Builder
+@NoArgsConstructor
+@AllArgsConstructor
 public class LanguageModel implements Comparable<LanguageModel> {
 
     private ModelProvider provider;

